shrinkwrap.py

```diff
@@ -19,6 +20,14 @@
 shlx = shlex.split
 
 
+@contextmanager
+def status(msg):
+    sys.stdout.write(f"    {msg} ... ")
+    sys.stdout.flush()
+    yield
+    print("done")
```

Comment:
It's not clear to me whether you wanted this contextmgr because it printed "done" at the end of a status-worthy task or because `print()` was coming up out-of-order without it.

If the former, carry on; if the latter, py3's `print` [has a flush param](https://docs.python.org/3/library/functions.html#print) that might make this moot.

Ninja edit:  after going through the rest of this pr, it does indeed seem you wanted a method for ack'ing the task completion.  i'll leave this here just in case i'm mis-understanding, but i think "carry on" is in order.

Reply:
you're right.  the context manager is there for
`say something is happening ...` <do it > `print done`